plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend

```diff
@@ -243,6 +266,95 @@ class TextualScenarioValidator extends AbstractTextualScenarioValidator {
 			)
 		}
 	}
+	
+	/*
+	 * Expression shall not be empty
+	 */
+	@Check
+	def checkCombinedFragmentEmptyExpression(CombinedFragment combinedFragment) {
+		if (combinedFragment.expression == null || combinedFragment.expression.isEmpty) {
+			error(
+				'Expression can not be empty',

```

Comment:
for state fragments, if an element is missing (null in this case), the validation sounds something like "please insert .. ". we could use the same pattern or we could change for state fragments :-?

Reply:
The expression is a String, is not a Capella object, to me, "expression can not be empty" sonds more appropriate that "insert...".

Reply:
okay, we could change the errors for state fragments in this case